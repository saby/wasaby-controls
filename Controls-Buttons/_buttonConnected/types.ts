import type { ObjectMeta } from 'Types/meta';

type TCommandOptions = Record<string, unknown>;

/**
 * Интерфейс объекта, описывающего действия.
 * @public
 */
export interface IActionConfig<RuntimeInterface extends object> {
    /**
     * Уникальный идентификатор действия
     * @remark это значение пишется в настройки действия во фрейме
     */
    id: string;

    /**
     * Человекопонятное название действия
     */
    title: string;
    /**
     * Категория действия
     */
    category?: string;
    /**
     * Иконка действия
     */
    icon?: string;
    /**
     * Алиас до модуля-исполнителя команды
     * @interface
     */
    commandName?: string;
    /**
     * Опции, которые пережаются в commandName первым аргументом
     */
    commandOptions?: TCommandOptions;
    /**
     * Мета-тип атрибутов команды
     */
    propsMeta?: ObjectMeta<RuntimeInterface>;
}

/**
 * Интерфейс модуля-исполнителя команды
 */
export interface IActionExecutor {
    /**
     * Метод, запускающий выполнение действия
     * @param commandOptions опции действия, заданные в его описании
     * @param editorProps опции действия, настроенные в конструкторе через {@link Controls-editors/propertyGrid:PropertyGrid PropertyGrid}
     */
    execute(commandOptions: TCommandOptions, editorProps: Record<string, unknown>);
}
