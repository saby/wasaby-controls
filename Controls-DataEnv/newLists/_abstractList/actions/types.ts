import type * as marker from './types/marker';
import type * as operationsPanel from './types/operationsPanel';
import type * as root from './types/root';
import type * as search from './types/search';
import type * as selection from './types/selection';
import type * as source from './types/source';
import type * as expandCollapse from './types/expandCollapse';
import type * as filter from './types/filter';

import type * as interactorCore from './types/interactorCore';

export type {
    marker,
    operationsPanel,
    root,
    search,
    selection,
    source,
    expandCollapse,
    filter,
    interactorCore,
};

/**
 * Тип действия, доступного в любом списке, независимо от типа ViewModel, к которой он подключен (web/mobile).
 */
export type TAnyAbstractAction =
    | source.TAnySourceAction
    | expandCollapse.TAnyExpandCollapseAction
    | marker.TAnyMarkerAction
    | filter.TAnyFilterAction
    | selection.TAnySelectionAction
    | operationsPanel.TAnyOperationsPanelAction
    | root.TAnyRootAction
    | search.TAnySearchAction
    | interactorCore.TAnyInteractorCoreAction;
