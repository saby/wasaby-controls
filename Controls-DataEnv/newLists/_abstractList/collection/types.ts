export { TCollectionType } from './types/TCollectionType';
