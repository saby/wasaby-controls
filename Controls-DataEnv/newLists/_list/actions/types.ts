import type { TAbstractListActions } from 'Controls-DataEnv/abstractList';

import type * as marker from './types/marker';
import type * as operationsPanel from './types/operationsPanel';
import type * as root from './types/root';
import type * as search from './types/search';
import type * as selection from './types/selection';
import type * as items from './types/items';
import type * as filter from './types/filter';
import type * as source from './types/source';
import type * as expandCollapse from './types/expandCollapse';
import type * as complexUpdate from './types/complexUpdate';

export type {
    marker,
    operationsPanel,
    root,
    expandCollapse,
    search,
    selection,
    items,
    filter,
    source,
    complexUpdate,
};

/**
 * Тип действия, доступного в WEB списке.
 */
export type TAnyListAction =
    | TAbstractListActions.TAnyAbstractAction
    | expandCollapse.TAnyExpandCollapseAction
    | marker.TAnyMarkerAction
    | selection.TAnySelectionAction
    | operationsPanel.TAnyOperationsPanelAction
    | root.TAnyRootAction
    | search.TAnySearchAction
    | items.TAnyItemsAction
    | source.TAnySourceAction
    | filter.TAnyFilterAction
    | complexUpdate.TAnyComplexUpdateAction;
