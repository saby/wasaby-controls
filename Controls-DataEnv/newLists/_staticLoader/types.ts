import type { IAbstractListState } from 'Controls-DataEnv/abstractList';

export type TState = Pick<IAbstractListState, 'viewMode' | 'isLatestInteractorVersion'>;

export interface TDependencyDescriptor<T extends keyof TState> {
    prop: T;
    value: TState[T][] | ((state: TState) => boolean);
}

export type TUI_Dependencies = Record<string, TDependencyDescriptor<keyof TState>[]>;
