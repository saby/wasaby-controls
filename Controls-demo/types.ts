import { IHeaderCell } from 'Controls/grid';
import { RecordSet } from 'Types/collection';

export interface IHeader extends IHeaderCell {
    title: string;
}

export type TItemsReadyCallback = (items: RecordSet) => void;
export type TRoot = string | number | null;

export interface INavigation {
    source: string;
    view: string;
    sourceConfig: {
        pageSize: number;
        page: number;
        hasMore?: boolean;
    };
    viewConfig?: {
        pagingMode?: string;
        maxCountValue?: number;
    };
}
