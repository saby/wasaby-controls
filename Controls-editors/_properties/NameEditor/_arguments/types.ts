import { ObjectMeta } from 'Meta/types';

const PARAMETERS_FIELD = 'parameters';
const RETURN_VALUE_FIELD = 'return_value';

interface IFunctionCallMeta {
    arguments: ObjectMeta | null;
    returnValue: ObjectMeta | null;
}

export { IFunctionCallMeta, PARAMETERS_FIELD, RETURN_VALUE_FIELD };
