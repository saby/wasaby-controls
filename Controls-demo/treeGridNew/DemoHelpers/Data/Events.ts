import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
