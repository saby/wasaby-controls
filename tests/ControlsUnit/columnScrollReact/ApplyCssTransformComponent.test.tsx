/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';

import { ApplyCssTransformComponent } from 'Controls/_columnScrollReact/ApplyCssTransformComponent';
import { EdgeState } from 'Controls/_columnScrollReact/common/types';

import { SELECTORS } from './SelectorsForTests';
import { renderStyleCompositeComponent } from './forTestsOnly/helpers';
import { AnimationSteps } from './forTestsOnly/animationSteps';

describe('ControlsUnit/columnScrollReact/ApplyCssTransformComponent', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    describe('mobile device.', () => {
        it('render only static and width dependent components.', () => {
            const { asFragment } = renderStyleCompositeComponent(
                <ApplyCssTransformComponent
                    selectors={SELECTORS}
                    isMobile={true}
                    fixedWidth={230}
                />,
                { container }
            );
            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('desktop device.', () => {
        describe('no animation.', () => {
            it('render static, width and position styles components.', () => {
                const { asFragment } = renderStyleCompositeComponent(
                    <ApplyCssTransformComponent
                        selectors={SELECTORS}
                        isMobile={false}
                        position={150}
                        fixedWidth={230}
                        previousPosition={130}
                        leftEdgeState={EdgeState.Invisible}
                        rightEdgeState={EdgeState.Invisible}
                        onAnimationEnd={() => {
                            return void 0;
                        }}
                    />,
                    { container }
                );
                expect(asFragment()).toMatchSnapshot();
            });
        });

        describe('animation.', () => {
            describe('static tests.', () => {
                AnimationSteps.forEach((params, index) => {
                    it(`${index}. ${params.from} -> ${params.to}. Edges: [${params.leftEdgeState}, ${params.rightEdgeState}]`, () => {
                        const { asFragment } = renderStyleCompositeComponent(
                            <ApplyCssTransformComponent
                                selectors={SELECTORS}
                                fixedWidth={230}
                                isMobile={false}
                                previousPosition={params.from}
                                position={params.to}
                                leftEdgeState={params.leftEdgeState}
                                rightEdgeState={params.rightEdgeState}
                                onAnimationEnd={jest.fn()}
                            />,
                            { container }
                        );
                        expect(asFragment()).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
