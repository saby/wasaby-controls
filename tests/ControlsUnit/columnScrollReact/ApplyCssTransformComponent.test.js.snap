// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. animation. static tests. 0. 0 -> 100. Edges: [AnimatedFromVisibleToInvisible, AnimatedInvisible] 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 700px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__smooth-styles"
  >
    @keyframes scrollAnimation_default {from{transform:translate3d(0px,0,0);}to{transform:translate3d(-100px,0,0);}}@keyframes fixationAnimation_default {from{transform:translate3d(0px,0,0);}to{transform:translate3d(100px,0,0);}}@keyframes fixationAnimation_right {from{transform:translate3d(-400px,0,0);}to{transform:translate3d(-300px,0,0);}}.js-controls-ColumnScrollReact_rootElement_guid123 { animation-name: scrollAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed_guid123 { animation-name: fixationAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { animation-name: fixationAnimation_right;animation-duration: 0.25s; }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(0px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(0px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-400px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. animation. static tests. 1. 100 -> 200. Edges: [AnimatedInvisible, AnimatedInvisible] 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 700px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__smooth-styles"
  >
    @keyframes scrollAnimation_default {from{transform:translate3d(-100px,0,0);}to{transform:translate3d(-200px,0,0);}}@keyframes fixationAnimation_default {from{transform:translate3d(100px,0,0);}to{transform:translate3d(200px,0,0);}}@keyframes fixationAnimation_right {from{transform:translate3d(-300px,0,0);}to{transform:translate3d(-200px,0,0);}}.js-controls-ColumnScrollReact_rootElement_guid123 { animation-name: scrollAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed_guid123 { animation-name: fixationAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { animation-name: fixationAnimation_right;animation-duration: 0.25s; }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(-100px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(100px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-300px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. animation. static tests. 2. 200 -> 300. Edges: [AnimatedInvisible, AnimatedFromInvisibleToVisible] 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 700px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__smooth-styles"
  >
    @keyframes scrollAnimation_default {from{transform:translate3d(-200px,0,0);}to{transform:translate3d(-300px,0,0);}}@keyframes fixationAnimation_default {from{transform:translate3d(200px,0,0);}to{transform:translate3d(300px,0,0);}}@keyframes fixationAnimation_right {from{transform:translate3d(-200px,0,0);}to{transform:translate3d(-100px,0,0);}}.js-controls-ColumnScrollReact_rootElement_guid123 { animation-name: scrollAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed_guid123 { animation-name: fixationAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { animation-name: fixationAnimation_right;animation-duration: 0.25s; }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(-200px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(200px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-200px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. animation. static tests. 3. 300 -> 200. Edges: [AnimatedInvisible, AnimatedFromVisibleToInvisible] 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 700px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__smooth-styles"
  >
    @keyframes scrollAnimation_default {from{transform:translate3d(-300px,0,0);}to{transform:translate3d(-200px,0,0);}}@keyframes fixationAnimation_default {from{transform:translate3d(300px,0,0);}to{transform:translate3d(200px,0,0);}}@keyframes fixationAnimation_right {from{transform:translate3d(-100px,0,0);}to{transform:translate3d(-200px,0,0);}}.js-controls-ColumnScrollReact_rootElement_guid123 { animation-name: scrollAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed_guid123 { animation-name: fixationAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { animation-name: fixationAnimation_right;animation-duration: 0.25s; }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(-300px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(300px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-100px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. animation. static tests. 4. 200 -> 100. Edges: [AnimatedInvisible, AnimatedInvisible] 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 700px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__smooth-styles"
  >
    @keyframes scrollAnimation_default {from{transform:translate3d(-200px,0,0);}to{transform:translate3d(-100px,0,0);}}@keyframes fixationAnimation_default {from{transform:translate3d(200px,0,0);}to{transform:translate3d(100px,0,0);}}@keyframes fixationAnimation_right {from{transform:translate3d(-200px,0,0);}to{transform:translate3d(-300px,0,0);}}.js-controls-ColumnScrollReact_rootElement_guid123 { animation-name: scrollAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed_guid123 { animation-name: fixationAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { animation-name: fixationAnimation_right;animation-duration: 0.25s; }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(-200px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(200px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-200px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. animation. static tests. 5. 100 -> 0. Edges: [AnimatedFromInvisibleToVisible, AnimatedInvisible] 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 700px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__smooth-styles"
  >
    @keyframes scrollAnimation_default {from{transform:translate3d(-100px,0,0);}to{transform:translate3d(0px,0,0);}}@keyframes fixationAnimation_default {from{transform:translate3d(100px,0,0);}to{transform:translate3d(0px,0,0);}}@keyframes fixationAnimation_right {from{transform:translate3d(-300px,0,0);}to{transform:translate3d(-400px,0,0);}}.js-controls-ColumnScrollReact_rootElement_guid123 { animation-name: scrollAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed_guid123 { animation-name: fixationAnimation_default;animation-duration: 0.25s; }.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { animation-name: fixationAnimation_right;animation-duration: 0.25s; }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(-100px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(100px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-300px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent desktop device. no animation. render static, width and position styles components. 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 600px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
  <style
    data-qa="ColumnScrollReact__position-styles"
  >
    .js-controls-ColumnScrollReact_rootElement_guid123 {transform:translate3d(-150px,0,0);}.js-controls-ColumnScrollReact_fixed_guid123 {transform:translate3d(150px,0,0);}.js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 {transform:translate3d(-1050px,0,0);
  </style>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ApplyCssTransformComponent mobile device. render only static and width dependent components. 1`] = `
<DocumentFragment>
  <style
    data-qa="ColumnScrollReact__static-styles"
  >
    .js-controls-ColumnScrollReact_fixed_guid123 {transform: translate3d(0, 0, 0) rotate(0) skewX(0) skewY(0) scaleX(1) scaleY(1);will-change: transform;}.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed_guid123,.js-controls-ColumnScrollReact_hideFixedElements_guid123 .js-controls-ColumnScrollReact_fixed-toRightEdge_guid123 { visibility: hidden; }
  </style>
  <style
    data-qa="ColumnScrollReact__width-styles"
  >
    .js-controls-ColumnScrollReact_offsetForFixed_guid123 { left: 230px; } .js-controls-ColumnScrollReact_offsetForEndFixed_guid123 { right: 0px; }.js-controls-ColumnScrollReact_stretchedToViewPortElement_guid123 { width: 500px; }.js-controls-ColumnScrollReact_rootElement_guid123 .controls-Grid__header-cell_withColumnScrollArrows { padding-bottom: calc( var(--inline_height_ArrowButton) + 2 * var(--offset_2xs) ); }
  </style>
</DocumentFragment>
`;
